lib/business_logic/dog_list_provider.dart

```diff
@@ -3,19 +3,20 @@ import 'package:array_names/data/repositories/repository.dart';
 import 'package:array_names/service_locator.dart';
 import 'package:flutter_riverpod/flutter_riverpod.dart';
 
-final getDogListProvider = StateNotifierProvider(
+final getDogListProvider = StateNotifierProvider<DogListProvider, List<Breed>>(
   (ref) {
     final repository = ref.read(repositoryProvider);
-    return DogListProvider(repository);
+    final provider = DogListProvider(repository);
+    return provider;
   },
 );
 
 class DogListProvider extends StateNotifier<List<Breed>> {
   final Repository repo;
 
   DogListProvider(this.repo) : super([]);
-  getDogs() async {
+  void getDogs() async {
```

Comment:
JFYI, could be refactored as
`
void getDogs() async => state = await repo.fetchDogs()
`